framework/components/ecl-social-media-links/ecl-social-media-links.twig

```diff
@@ -1,9 +1,77 @@
-<div class="social-media-links {{modifier}}">
-  <p>Follow the latest progress and learn more about getting involved.</p>
-  <span class="social-media-links__label">Share this blog post:</span>
-  <ul class="social-media-links__list">
-    <li class="social-media-links__item"><a href="http://facebook.com" title="Facebook" class="icon-social icon-social--facebook">Facebook</a></li>
-    <li class="social-media-links__item"><a href="http://twitter.com" title="Twitter" class="icon-social icon-social--twitter">Twitter</a></li>
-    <li class="social-media-links__item social-media-links__item--noicon"><a href="#">Other social networks.</a></li>
+{#
+  Parameters:
+    - "variant" (string): can be empty or 'horizontal' (default: empty)
+    - "has_text" (bool): display text before links or not (default: true)
+    - "text" (string): additional text displayed before the links (default: 'View European Commission on:')
+    - "links" (array): [{
+        "href" (string): target of the link (default: empty)
+        "label" (string): label of the link (default: empty)
+        "title" (string): title of the link (default: empty)
+        "variant" (string): variant of the social network. Should be one of the variant listed in Social icons component (default: empty)
+      }]
+      - "more" {object}: {
+        - "href" (string): target of the "more" link (default: empty)
+        - "label" (string): label of the "more" link (default: 'Other social networks')
+      }
+    - "extra_classes" (string): extra CSS classes to be added
+    - "extra_attributes" (array): extra attributes classes (optional, format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
+  Blocks:
+    - "text_block": instead of providing additional text as a string, you can also embed the template and fill the "text" block directly
+#}
+
+{# Internal properties #}
+
+{% set _has_text = has_text|default(false) %}
+{% set _text = text|default('View European Commission on:') %}
+{% set _more = more|default({
+  label: more.label|default('Other social networks'),
+  href: more.href|default('')
+}) %}
+{% set _css_class = 'ecl-social-media-link' %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+
+{% if variant is defined and variant is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ 'ecl-social-media-link--' ~ variant %}
+{% endif %}
+
+{% if extra_classes is defined %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print result #}
+
+<div class="{{ _css_class }}" {{ _extra_attributes }}>
+  {% block text_block %}
```

Comment:
The logic here could be refined (`has_text` could be replaced by a test on `text`?)

Reply:
That was my first intention, but as text has a default value I didn't know how to test it